src/modules/worlds/sagas.spec.ts

```diff
@@ -63,11 +66,68 @@ describe('when handling the request action to fetch worlds stats for a wallet',
   })
 })
 
-describe('when the wallet connects', () => {
-  it('should put the action to fetch worlds stats for the connected wallet address', () => {
-    return expectSaga(worldsSaga)
-      .put(fetchWorldsWalletStatsRequest(address))
-      .dispatch(connectWalletSuccess({ address } as Wallet))
-      .silentRun()
+describe('when handling the connect wallet success action', () => {
+  describe('when there is no address in the store', () => {
+    it('should not put the request action to fetch worlds wallet stats', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), undefined]])
+        .not.put(fetchWorldsWalletStatsRequest(address))
+        .dispatch(connectWalletSuccess({} as Wallet))
+        .silentRun()
+    })
+  })
+
+  describe('when there is an address in the store', () => {
+    it('should put the request action to fetch worlds wallet stats with the stored address', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), address]])
+        .put(fetchWorldsWalletStatsRequest(address))
+        .dispatch(connectWalletSuccess({} as Wallet))
+        .silentRun()
+    })
+  })
+})
+
+describe('when handling the deploy to world success action', () => {
+  describe('when there is no address in the store', () => {
+    it('should not put the request action to fetch worlds wallet stats', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), undefined]])
+        .not.put(fetchWorldsWalletStatsRequest(address))
+        .dispatch(deployToWorldSuccess({} as Deployment))
+        .silentRun()
+    })
+  })
+
+  describe('when there is an address in the store', () => {
+    it('should put the request action to fetch worlds wallet stats with the stored address', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), address]])
+        .put(fetchWorldsWalletStatsRequest(address))
+        .dispatch(deployToWorldSuccess({} as Deployment))
+        .silentRun()
+    })
+  })
+})
+
+describe('when handling the clear deployment success action', () => {
+  describe('when there is no address in the store', () => {
+    it('should not put the request action to fetch worlds wallet stats', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), undefined]])
+        .not.put(fetchWorldsWalletStatsRequest(address))
+        .dispatch(clearDeploymentSuccess(''))
+        .silentRun()
+    })
+  })
+
+  describe('when there is an address in the store', () => {
+    it('should put the request action to fetch worlds wallet stats with the stored address', () => {
+      return expectSaga(worldsSaga)
+        .provide([[select(getAddress), address]])
```

Comment:
As `expectSaga` runs the saga until completion and, when testing `handleActionsThatTriggerFetchWorldsWalletStatsRequest`, the `fetchWorldsWalletStatsRequest` action is put and, taking into consideration that the sagas also handles that action, the `fetchWorldsWalletStatsRequest` should be ran as well. To prevent this, we can mock the put:
```suggestion
        .provide([[select(getAddress), address], [put(fetchWorldsWalletStatsRequest(address)), undefined]])
```

Reply:
I'm certain that is unnecessary due to the nature of sagas.

`fetchWorldsWalletStatsRequest` will be handled by the saga yes, but it will not proceed on any undeclared effect.

For example, it will run and stop were said in the code:

```js
function* handlefetchWorldsWalletStatsRequest(action: FetchWalletWorldsStatsRequestAction) {
  const { address } = action.payload

  try {
    // Stops here given that the call is not declared in the test
    const stats: WorldsWalletStats | null = yield call([WorldsAPIContent, WorldsAPIContent.fetchWalletStats], address)
```

Reply:
Besides if I set the request put in the .provide, it will not call the .put that comes afterwards, making the test not declaratively useful given that the things tested are what comes after the .provide

Reply:
I'm mistaken with this, updated it per your suggestion